connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSourceTask.java

```diff
@@ -181,7 +184,11 @@ public void execute() {
      */
     private boolean sendRecords() {
         int processed = 0;
-        for (final SourceRecord record : toSend) {
+        for (SourceRecord preTransformRecord : toSend) {
+            final SourceRecord record = transformationChain.apply(preTransformRecord);
+
+            if (record == null) continue;
```

Comment:
I imagine the most common usage of `commitRecord()` would be to track a high watermark of committed offset state. So I think it could cause incorrect state to invoke it for a record that's being filtered away, if it causes such a watermark to be advanced pre-maturely.

(Unless I'm misunderstanding that `commitRecord()` invocations would be in the order of the records that were created by the source connector).